WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewModel.swift

```diff
@@ -0,0 +1,88 @@
+import Foundation
+import UIKit
+import CoreData
+
+class BlogDashboardViewModel {
+    private weak var viewController: BlogDashboardViewController?
+
+    enum Section: CaseIterable {
+        case quickLinks
+        case posts
+    }
+
+    // FIXME: temporary placeholder
+    private let quickLinks = ["Quick Links"]
+    private let posts = ["Posts"]
+
+    typealias DataSource = UICollectionViewDiffableDataSource<Section, String>
+    typealias Snapshot = NSDiffableDataSourceSnapshot<Section, String>
+    typealias QuickLinksHostCell = HostCollectionViewCell<QuickLinksView>
+
+    private let managedObjectContext: NSManagedObjectContext
+    private let blog: Blog
+
+    private lazy var service: DashboardServiceRemote = {
+        let api = WordPressComRestApi.defaultApi(in: managedObjectContext,
+                                                 localeKey: WordPressComRestApi.LocaleKeyV2)
+
+        return DashboardServiceRemote(wordPressComRestApi: api)
+    }()
+
+    private lazy var dataSource: DataSource? = {
+        guard let viewController = viewController else {
+            return nil
+        }
+
+        return DataSource(collectionView: viewController.collectionView) { [unowned self] collectionView, indexPath, identifier in
+            switch identifier {
+            case self.quickLinks.first:
+                let cell = collectionView.dequeueReusableCell(withReuseIdentifier: QuickLinksHostCell.defaultReuseID, for: indexPath) as? QuickLinksHostCell
+                cell?.hostedView = QuickLinksView(title: self.quickLinks[indexPath.item])
+                return cell
+            case self.posts.first:
+                let cell = collectionView.dequeueReusableCell(withReuseIdentifier: DashboardPostsCardCell.defaultReuseID, for: indexPath) as? DashboardPostsCardCell
+                cell?.configure(viewController, blog: blog)
+                return cell
+            default:
+                break
+            }
+            return UICollectionViewCell()
+        }
+    }()
+
+    init(viewController: BlogDashboardViewController, managedObjectContext: NSManagedObjectContext = ContextManager.shared.mainContext, blog: Blog) {
+        self.viewController = viewController
+        self.managedObjectContext = managedObjectContext
+        self.blog = blog
+    }
+
+    /// Call the API to return cards for the current blog
+    func start() {
+        viewController?.showLoading()
+        applySnapshotForInitialData()
+
+        service.fetch(cards: ["posts", "todays_stats"], forBlogID: blog.dotComID as! Int, success: { [weak self] _ in
```

Comment:
What do you think of using guard let to cast `blog.dotComID` as an Int?

Reply:
Fixed in 2898f813658cac9559bcd35558e16b7ae41b744e